### PR TITLE
164798913 Users should be able to comment on articles

### PR DESCRIPTION
**Pull request by davidokonji**
#### Type of change
- [x]  add tests for endpoints
- [x] add a route to create a comment on an article
- [x] add a route to create threaded comments
- [x] documented endpoints created
Screenshot 2019-04-05 at 17 09 53
>- User makes threaded comments
Screenshot 2019-04-05 at 17 10 17

#### Checklist:
- [x]  add tests for endpoints
- [x] add a route to create a comment on an article
- [x] add a route to create threaded comments
- [x] documented endpoints created
#### Story (PT)
[Delivers #164798913]

### CODE REVIEW

File: server/controllers/comment.controller.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import dotenv from 'dotenv';
+import db from '../models';
+import { Response } from '../utils';
+
+dotenv.config();
```
Comment (waleCloud) [inline server/controllers/comment.controller.js:5]: why is this here?

File: server/models/comments.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    comments.belongsTo(models.Article, {
+      foreignKey: 'articleid',
+      otherKey: 'userid',
+      through: 'comments',
+      as: 'articleComments',
+      timestamps: false
+    });
+  };
+  return comments;
+};
```
Comment (Chiazokam) [inline server/models/comments.js]: Everything looks good but for the EOL. Never forget. Good practice.

Comment (davidokonji) [reply server/models/comments.js]: Ok thanks

File: server/routes/comment.route.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import controllers from '../controllers';
+import { tryCatch, checkComments } from '../utils';
+import { signInAuth } from '../utils/users/permissions.util';
+
+const { CommentController } = controllers;
+const { createComments, threadedComment } = CommentController;
+
+const router = new Router();
+
+router.post('/:slug/comment', [checkComments, signInAuth], tryCatch(createComments));
```
Comment (chukwudidu) [inline server/routes/comment.route.js]: Good job grouping the middleware in an array. Would help for readability.

**Review by TheDrizzyWay — COMMENTED** [review]
Looks good to me.

File: server/test/comment.spec.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+let userToken;
+before(async () => {
+  const userResponse = await chai
+    .request(app)
+    .post('/api/v1/auth/login')
+    .send({
+      email: 'mofe@okoro.com',
+      password: 'soldier'
+    });
+  userToken = userResponse.body.data.token;
```
Comment (chukwudidu) [inline server/test/comment.spec.js]: I think maybe destructuring the token from the `body` would be ES6 compliant

**Review by chukwudidu — COMMENTED** [review]
Good job

File: server/routes/comment.route.js (hunk 3 above)
Comment (davidokonji) [reply server/routes/comment.route.js]: Thank you

File: server/test/comment.spec.js (hunk 4 above)
Comment (davidokonji) [reply server/test/comment.spec.js]: Ok would look into that now, thanks

File: server/test/comment.spec.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+});
+describe('User post comment route', () => {
```
Comment (geekdanitek) [inline server/test/comment.spec.js:19]: Good job handling all test cases possible.

File: server/utils/comments/comments.util.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import validator from 'validator';
```
Comment (geekdanitek) [inline server/utils/comments/comments.util.js:1]: I like the fact that your variable naming is descriptive which makes anyone reading the code to understand the function of the variable.

**Review by geekdanitek — APPROVED** [review]

File: server/migrations/20190404145044-create-comments.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
+'use strict';
```
Comment (samorahno) [inline server/migrations/20190404145044-create-comments.js]: Good job @davidokonji , you might consider removing this line to prevent some strict pattern in Javascript. or what do you think?

**Review by samorahno — COMMENTED** [review]
Neat structure @davidokonji, I love the fact that I was able to flow along with the code.

File: server/migrations/20190404145044-create-comments.js (hunk 7 above)
Comment (davidokonji) [reply server/migrations/20190404145044-create-comments.js]: Ok would work on that, thanks

File: server/test/comment.spec.js (hunk 5 above)
Comment (davidokonji) [reply server/test/comment.spec.js:19]: Thank you

File: server/controllers/comment.controller.js (hunk 1 above)
Comment (davidokonji) [reply server/controllers/comment.controller.js:5]: @waleCloud  used it in order to make use of environmental variables, would find an alternate way. Thanks
